assets/src/edit-story/components/canvas/layout.js

```diff
@@ -63,26 +64,45 @@ const Layer = styled.section`
   ${({ zIndex }) => (typeof zIndex === 'number' ? `z-index: ${zIndex};` : '')}
 
   display: grid;
+  /*
+    . = empty space
+    h = header
+    b = back navigation
+    f = forward navigation
+    p = canvas page
+    m = page action menu
+    c = thumbnail carousel
+
+    Also note that we need to specify all the widths and heights
+    even though some of the elements could just use the size that
+    the element takes up. This is because we reuse this grid in 3
+    different layers on top of each other and some elements are
+    missing in some layers, but we still need them to align perfectly.
+  */
   grid:
-    'head      head      head      head      head    ' ${HEADER_HEIGHT}px
-    '.         .         .         .         .       ' minmax(16px, 1fr)
-    '.         prev      page      next      .       ' var(
-      --fullbleed-height-px
-    )
-    '.         .         menu      .         .       ' ${MENU_HEIGHT}px
-    '.         .         .         .         .       ' 1fr
-    'carousel  carousel  carousel  carousel  carousel' ${CAROUSEL_HEIGHT}px
-    / 1fr ${PAGE_NAV_WIDTH}px var(--fullbleed-width-px)
-    ${PAGE_NAV_WIDTH}px 1fr;
+    'h h h h h h h' ${HEADER_HEIGHT}px
+    '. . . . . . .' minmax(16px, 1fr)
+    '. b . p . f .' var(--fullbleed-height-px)
+    '. . . . . . .' ${MENU_GAP}px
+    'm m m m m m m' ${MENU_HEIGHT}px
+    '. . . . . . .' 1fr
+    'c c c c c c c' ${CAROUSEL_HEIGHT}px
```

Comment:
Yay! Almost looks like a room full of monsters in [Nethack](https://www.nethack.org/) 😍 .